### PR TITLE
Modified LinkedLists structure and methods for custom comparatorFunction

### PR DESCRIPTION
**Pull request by naman-modi**
According to issue ["Add a comparator #1"](https://github.com/yashshah1/yard-ts/issues/1), wherein it was mentioned that "==" is extremely unscalable, I have modified the structures of linked-list by adding a function: <br>
`private _compFunc: (x: any, y: any) => boolean;` <br>
and setting it to default "==" in the constructor. <br>
So I have accordingly changed the methods,<ul><li><i>deleteFirstOccurence</i></li><li><i>deleteAllOccurences</i></li></ul> such that the user can pass the customized comparator function as an argument to the above methods.

### CODE REVIEW

File: src/linked-list/doubly-linked-list/DoublyLinkedList.ts [hunk 1]
```diff
@@ -17,10 +17,16 @@ export default class DoublyLinkedList {
   private _count: number;
   private _head: DoublyLinkedListNode | null;
   private _tail: DoublyLinkedListNode | null;
+  private _compFunc: (x: any, y: any) => boolean;
+
   constructor() {
```
Comment (yashshah1) [inline src/linked-list/doubly-linked-list/DoublyLinkedList.ts:22]: Add a param here, which the user can pass
`constructor(compFunc?: (x: any, y: any) => boolean)`

File: src/linked-list/doubly-linked-list/DoublyLinkedList.ts [hunk 2]
```diff
@@ -17,10 +17,16 @@ export default class DoublyLinkedList {
...
     this._head = null;
     this._tail = null;
     this._count = 0;
+    this._compFunc = (x: any, y: any): boolean => {
```
Comment (yashshah1) [inline src/linked-list/doubly-linked-list/DoublyLinkedList.ts:26]: Similarly, this doesn't account into whether the user has passed on or no.
Maybe, this? `this._compFunc = compFunc ? compFunc : .......`

File: src/linked-list/doubly-linked-list/DoublyLinkedList.ts (hunk 1 above)
Comment (naman-modi) [reply src/linked-list/doubly-linked-list/DoublyLinkedList.ts:22]: So here, the declaration of `compFunction` doesn't change, as we expect the user to pass a function that takes 2 arguments as input and returns a boolean value. <br>
I will make changes to the constructor.

File: src/linked-list/doubly-linked-list/DoublyLinkedList.ts (hunk 2 above)
Comment (naman-modi) [reply src/linked-list/doubly-linked-list/DoublyLinkedList.ts:26]: Similarly, I will make changes to the constructor.

**Comment by naman-modi** [comment]
I have done the required changes, that is, modified the constructor to accept customized ComparatorFunctions from the user. Please review.

Regards,
[Naman](https://github.com/naman-modi)

**Comment by yashshah1** [comment]
lgtm

**Comment by yashshah1** [comment]
@all-contributors please add @naman-modi for code

**Pull request merged**